src/NHibernate/Async/Loader/Loader.cs

```diff
@@ -706,7 +706,9 @@ protected virtual Task<object[]> GetResultRowAsync(Object[] row, DbDataReader rs
 			object id = key.Identifier;
 
 			// Get the persister for the _subclass_
-			ILoadable persister = (ILoadable)Factory.GetEntityPersister(instanceClass);
+			ILoadable persister = ReferenceEquals(instanceClass, rootPersister.EntityName)
```

Comment:
Are you sure that should be ReferenceEquals?

Reply:
I think it's pretty safe and most effective.

I've verified that the same instance is retrieved when subclass lookup `persister.GetSubclassForDiscriminatorValue` is performed for base subclass discriminator value. And from what I see all entity names  come from `PersistentClass.EntityName'

Maybe I should add an additional comment to `GetInstanceClass`. Something like: if entity name is dynamically calculated and might be equal to persister.EntityName please return perster.EntityName to avoid additional persister lookup in `GetResultRow`.

Or just let me know if you think it's better to use full string comparison here :)

Reply:
Personally I am hard to convince about doing such micro-optimisation. Here we do not actually care about the string having the same reference but about them being the same according to default equality.
Default string equality is already quite optimized. I do not think it is worth trying to be "smarter", at the risk of causing more inefficiency with some later code change, even if "protected" by a comment asking for coding in a special way due to this micro-optimization here.

Reply:
Ok (though I don't think this code can cause more inefficiency  than before)  What about comparer? :)  Ordinal? Default (culture specific)?

I see that `SessionFactoryImpl.entityPersisters` is created with default culture specific comparer. Is it really right? I think it should be either Ordinal or Invariant (I'm voting for  Ordinal for entity names)

Reply:
Oups.. For some reasons I thought that string.Equals do by default culture specific comparison.. Sorry for all the fuss :)